spec/queries/get_activity_log_events_spec.rb

```diff
@@ -0,0 +1,122 @@
+require 'rails_helper'
+
+RSpec.describe GetActivityLogEvents, with_audited: true do
+  around do |example|
+    @now = Time.zone.local(2020, 2, 11)
+    Timecop.travel(@now) { example.run }
+  end
+
+  let(:provider_user) { create(:provider_user, :with_two_providers) }
+  let(:application_choices_for_provider_user) { GetApplicationChoicesForProviders.call(providers: provider_user.providers) }
+  let(:service) { GetActivityLogEvents.new(application_choices: application_choices_for_provider_user) }
+
+  let(:course_provider_a) { create(:course, provider: provider_user.providers.first) }
+  let(:course_provider_b) { create(:course, provider: provider_user.providers.second) }
+  let(:course_unrelated) { create(:course) }
+  let(:ratified_course_provider_b) { create(:course, accredited_provider: provider_user.providers.second) }
+  let(:ratified_course_unrelated) { create(:course, accredited_provider: create(:provider)) }
+
+  def create_application_choice_for_course(course)
+    course_option = course.course_options.first
+    course_option ||= create(:course_option, course: course)
+    create(:application_choice, :awaiting_provider_decision, course_option: course_option)
+  end
+
+  def create_audit_for_application_choice(application_choice, changes: {})
+    create(:application_choice_audit, user: provider_user, application_choice: application_choice, changes: changes)
+  end
+
+  describe '#call' do
+    it 'raises an error unless application_choices responds_to #to_sql ' do
+      array = 2.times.map { create_application_choice_for_course(course_provider_a) }
+      expect { GetActivityLogEvents.new(application_choices: array).call }.to raise_error(NoMethodError)
+    end
+
+    it 'returns an empty array if no audits are found' do
+      expect(service.call).to eq([])
+    end
+
+    it 'returns objects responding to the required attributes' do
+      choice = create_application_choice_for_course course_provider_a
+      create_audit_for_application_choice choice
+
+      result = service.call
+
+      expect(result.count).to eq(1)
+
+      %i[created_at auditable].each do |attr|
```

Comment:
We should probably call this `auditable` `application_choice` instead, to avoid confusion

Reply:
`application_choice` would be more readable, yes, but the query returns `Audited::Audit` objects which already expose `.auditable` as method for referencing the polymorphic `auditable_type` and `auditable_id` data. For efficiency I've decided not to decorate or manipulate the results so that they have an `application_choice`.

The only code which will need to call `.auditable` will be inside the view or the view_component presenting the elements of the list (which are audit events), so I think this is a reasonable compromise.